tests/components/hdmi_cec/test_media_player.py

```diff
@@ -0,0 +1,480 @@
+
+
+async def test_service_off(hass, create_hdmi_network, create_cec_entity, assert_state):
+    """Test that media_player triggers on `off` service."""
+    hdmi_network = await create_hdmi_network({"platform": "media_player"})
+    mock_hdmi_device = MockHDMIDevice(logical_address=3)
+    await create_cec_entity(hdmi_network, mock_hdmi_device)
+    state = hass.states.get("media_player.hdmi_3")
+    assert state.state != STATE_OFF
+
+    await hass.services.async_call(
+        MEDIA_PLAYER_DOMAIN,
+        SERVICE_TURN_OFF,
+        {ATTR_ENTITY_ID: "media_player.hdmi_3"},
+        blocking=True,
+    )
+
+    mock_hdmi_device.turn_off.assert_called_once_with()
+
+    state = hass.states.get("media_player.hdmi_3")
+    assert_state(state.state, STATE_OFF)
+
+
+@pytest.mark.parametrize(
+    "type_id,expected_features",
+    [
+        (TYPE_TV, (MPEF.TURN_ON, MPEF.TURN_OFF)),
+        (
+            TYPE_RECORDER,
+            (
+                MPEF.TURN_ON,
+                MPEF.TURN_OFF,
+                MPEF.PLAY,
+                MPEF.PAUSE,
+                MPEF.STOP,
+                MPEF.PREVIOUS_TRACK,
+                MPEF.NEXT_TRACK,
+            ),
+        ),
+        (TYPE_UNKNOWN, (MPEF.TURN_ON, MPEF.TURN_OFF)),
+        pytest.param(
+            TYPE_TUNER,
+            (
+                MPEF.TURN_ON,
+                MPEF.TURN_OFF,
+                MPEF.PLAY,
+                MPEF.PAUSE,
+                MPEF.STOP,
+            ),
+            marks=pytest.mark.xfail(
+                reason="Checking for the wrong attribute, should be checking `type_id`, is checking `type`."
+            ),
+        ),
+        pytest.param(
+            TYPE_PLAYBACK,
+            (
+                MPEF.TURN_ON,
+                MPEF.TURN_OFF,
+                MPEF.PLAY,
+                MPEF.PAUSE,
+                MPEF.STOP,
+                MPEF.PREVIOUS_TRACK,
+                MPEF.NEXT_TRACK,
+            ),
+            marks=pytest.mark.xfail(
+                reason="Checking for the wrong attribute, should be checking `type_id`, is checking `type`."
+            ),
+        ),
+        (
+            TYPE_AUDIO,
+            (
+                MPEF.TURN_ON,
+                MPEF.TURN_OFF,
+                MPEF.VOLUME_STEP,
+                MPEF.VOLUME_MUTE,
+            ),
+        ),
+    ],
+)
+async def test_supported_features(
+    hass, create_hdmi_network, create_cec_entity, type_id, expected_features
+):
+    """Test that features load as expected."""
+    hdmi_network = await create_hdmi_network({"platform": "media_player"})
+    mock_hdmi_device = MockHDMIDevice(
+        logical_address=3, type=type_id, type_name=DEVICE_TYPE_NAMES[type_id]
+    )
+    await create_cec_entity(hdmi_network, mock_hdmi_device)
+
+    state = hass.states.get("media_player.hdmi_3")
+    supported_features = state.attributes["supported_features"]
+    xfail = False
+    for feature in expected_features:
+        if feature == MPEF.PLAY:
```

Comment:
We want to avoid conditional logic in tests. If possible use parametrization or separate tests instead. In this case we can address it later as I assume we're aiming at fixing the fail cases.

Reply:
I get that. I also don't really like solution, but I couldn't think of a cleaner one. I'll look at the code and see if I can find a better solution :)